File: Mathematical Algorithms/Tribonacci Numbers/Tribonacci.py
def printTrib(n):
    dp = [0] * n
    dp[0] = dp[1] = 0
    dp[2] = 1

    for i in range(3, n):
        dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3]

    for i in range(0, n):
        print(dp[i], " ", end="")

File: Mathematical Algorithms/Tribonacci Numbers/Tribonacci.cpp
#include <iostream>
using namespace std;

int printTrib(int n) {
    int dp[n];
    dp[0] = dp[1] = 0;
    dp[2] = 1;

    for (int i = 3; i < n; i++)
        dp[i] = dp[i - 1] +
                dp[i - 2] +
                dp[i - 3];

    for (int i = 0; i < n; i++)
        cout << dp[i] << " ";
}